client/checkout/utils/upe.js

```diff
@@ -339,18 +326,17 @@ export const togglePaymentMethodForCountry = ( upeElement ) => {
 		wcpayCustomerData?.billing_country ||
 		'';
 
-	const upeContainer = document.querySelector(
-		'.payment_method_woocommerce_payments_' + paymentMethodType
-	);
+	const upeContainer = upeElement.closest( 'li.wc_payment_method' );
```

Comment:
Traversing up trying to find the first parent which will be ready to be used so that we can properly hide/show the gateway